From ffc68cc6c2725147604d2a7f2fc84ba380839259 Mon Sep 17 00:00:00 2001
From: Nicolas Le Bayon <nicolas.le.bayon@st.com>
Date: Fri, 8 Jan 2021 14:18:42 +0100
Subject: [PATCH 0250/1695] stm32mp2: add RISAF device tree bindings

Adds device tree bindings for RISAF(Resource isolation slave unit
for address space protection (full version))

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git e2b52ff720a7f4dd37890b99c2be6b56f9177efc stm32mp2: add RISAF device tree bindings for subregions
c1928d1b1600e2497aebf839cc1a362208ddd350 stm32mp2: add RISAF device tree bindings]
Change-Id: If7a767ee79c68e3e843955577197fe3033b804b4
Signed-off-by: Gatien Chevallier <gatien.chevallier@st.com>
Signed-off-by: Nicolas Le Bayon <nicolas.le.bayon@st.com>
---
 .../include/dt-bindings/soc/stm32mp25-risaf.h | 36 +++++++++++++++++++
 1 file changed, 36 insertions(+)
 create mode 100644 core/include/dt-bindings/soc/stm32mp25-risaf.h

diff --git a/core/include/dt-bindings/soc/stm32mp25-risaf.h b/core/include/dt-bindings/soc/stm32mp25-risaf.h
new file mode 100644
index 000000000..7e1d4ca3e
--- /dev/null
+++ b/core/include/dt-bindings/soc/stm32mp25-risaf.h
@@ -0,0 +1,36 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause) */
+/*
+ * Copyright (C) 2020-2024, STMicroelectronics - All Rights Reserved
+ */
+
+#ifndef _DT_BINDINGS_STM32MP25_RISAF_H
+#define _DT_BINDINGS_STM32MP25_RISAF_H
+
+/* RISAF region IDs */
+#define RISAF_REG_ID(idx)	(idx)
+
+/* RISAF base region enable modes */
+#define RIF_BREN_DIS		0x0
+#define RIF_BREN_EN		0x1
+
+/* RISAF encryption modes */
+#define RIF_ENC_DIS		0x0
+#define RIF_ENC_EN		0x2
+
+#define DT_RISAF_EN_SHIFT	4
+#define DT_RISAF_SEC_SHIFT	5
+#define DT_RISAF_ENC_SHIFT	6
+#define DT_RISAF_PRIV_SHIFT	8
+#define DT_RISAF_READ_SHIFT	16
+#define DT_RISAF_WRITE_SHIFT	24
+
+#define RISAFPROT(risaf_region, cid_read_list, cid_write_list, cid_priv_list, sec, enc, enabled) \
+	(((cid_write_list) << DT_RISAF_WRITE_SHIFT) | \
+	 ((cid_read_list) << DT_RISAF_READ_SHIFT) | \
+	 ((cid_priv_list) << DT_RISAF_PRIV_SHIFT) | \
+	 ((enc) << DT_RISAF_ENC_SHIFT) | \
+	 ((sec) << DT_RISAF_SEC_SHIFT) | \
+	 ((enabled) << DT_RISAF_EN_SHIFT) | \
+	 (risaf_region))
+
+#endif /* _DT_BINDINGS_STM32MP25_RISAF_H */
-- 
2.47.2

